### PR TITLE
Riegeli: add support for decoding transposed chunks

### PR DESCRIPTION
**Pull request by schroederc**
The transposed chunk decoder is used for both RecordsMetadata chunks and
regular transposed record chunks.  Transposed chunks decompose records
into their component key-value entries and store alike entries (by type
and field) together to achieve better compression.  Transposed chunks
are decoded by interpreting a state machine encoded alongside the
compressed record data buffers.

Current testing involves reading golden data until an encoder is
implemented.

### CODE REVIEW

File: kythe/go/util/riegeli/golden_test.go [hunk 1]
```diff
@@ -19,24 +19,60 @@ package riegeli_test
...
 )
 
+func BenchmarkGoldenTestData(b *testing.B) {
+	for _, variant := range goldenRiegeliFileVariants {
+		file := strings.Join([]string{goldenRiegeliFilePrefix, variant, "riegeli"}, ".")
+		b.Run(variant, func(b *testing.B) { benchGoldenData(b, file) })
+	}
+}
+
+func benchGoldenData(b *testing.B, goldenRiegeliFile string) {
```
Comment (creachadair) [inline kythe/go/util/riegeli/golden_test.go]: Can you please document what this benchmark is benchmarking, at least in broad terms?

File: kythe/go/util/riegeli/reader.go [hunk 2]
```diff
@@ -22,36 +22,73 @@ import (
...
+			// ignore chunks with no records; even for unknown chunk types
 			continue
 		}
-		r.recordReader, err = newRecordChunkReader(c)
-		if err != nil {
-			return fmt.Errorf("bad record chunk: %v", err)
+
+		switch c.Header.ChunkType {
+		case fileSignatureChunkType:
+			// TODO (schroederc): verify
```
Comment (creachadair) [inline kythe/go/util/riegeli/reader.go]: What does ignoring this do? Will the results be valid if we skip the signatures?

File: kythe/go/util/riegeli/golden_test.go (hunk 1 above)
Comment (schroederc) [reply kythe/go/util/riegeli/golden_test.go]: ✓

File: kythe/go/util/riegeli/reader.go (hunk 2 above)
Comment (schroederc) [reply kythe/go/util/riegeli/reader.go]: Ignoring it means we could be reading a corrupt file, but skipping it won't be a problem for a valid file.

Comment (creachadair) [reply kythe/go/util/riegeli/reader.go]: It might be worth logging that. Although, if you're going to implement this soon anyway, it's not crucial.

Comment (schroederc) [reply kythe/go/util/riegeli/reader.go]: Better to just fix it.

File: kythe/go/util/riegeli/reader.go [hunk 3]
```diff
@@ -94,6 +96,15 @@ func (r *Reader) ensureRecordReader() error {
 	return nil
 }
 
+func verifySignature(c *chunk) error {
```
Comment (creachadair) [inline kythe/go/util/riegeli/reader.go]: I might just inline this, since it's shorter without. Your call, though.

**Pull request merged**